### PR TITLE
Support OL mapbox styles

### PR DESCRIPTION
**Pull request by gberaudo**
- [x] honor zIndex
- [x] investigate draw along line
- [x] investigate incorrect clipping
- [x] investigate decluttering (single layer)

### CODE REVIEW

**Comment by gberaudo** [comment]
image

**Comment by gberaudo** [comment]
Fixes #3.

File: package.json [hunk 1]
```diff
@@ -1,43 +1,57 @@
...
+    "eslint-import-resolver-typescript": "^2.5.0",
     "esm": "^3.2.25",
-    "jest": "^27.0.5",
+    "jest": "^27.2.3",
     "jest-canvas-mock": "^2.3.1",
-    "node-fetch": "^2.6.1",
-    "ts-jest": "^27.0.3",
-    "typescript": "^4.3.2",
-    "vite": "^2.3.8"
+    "lit": "2.0.0-rc.2",
```
Comment (fredj) [inline package.json]: lit 2.0.0 is out !

File: package.json [hunk 2]
```diff
@@ -1,43 +1,57 @@
...
   "devDependencies": {
-    "@babel/preset-env": "^7.14.7",
-    "@babel/preset-typescript": "^7.14.5",
-    "@types/jest": "^26.0.23",
-    "@types/ol": "^6.5.1",
+    "@babel/preset-env": "^7.15.6",
+    "@babel/preset-typescript": "^7.15.0",
+    "@openlayers/eslint-plugin": "^4.0.0",
+    "@types/jest": "^27.0.2",
+    "@types/ol": "^6.5.3",
```
Comment (fredj) [inline package.json]: Not needed with recent version of ol (untested)

**Review by fredj — APPROVED** [review]
looks very good, thanks !

**Pull request merged**